lib/asciidoctor-epub3/packager.rb

```diff
@@ -424,7 +425,7 @@ def package options = {}
 
     ::FileUtils.mkdir_p dest unless ::File.directory? dest
 
-    epub_file = ::File.expand_path %(#{doc.attr 'docname'}#{fmt == :kf8 ? '-kf8' : nil}.epub), dest
+    epub_file = ( fmt == :kf8 ? File.expand_path("#{File.basename(target, '.*')}-kdf8.epub", dest) : target )
```

Comment:
When generate epub, `target` is the final filename. When generate mobi, first generate `target-kdf8.epub` in dest directory, then `kindlegen -o target.mobi target-kdf8.epub`.